auth/sdk/src/main/kotlin/com/walletconnect/auth/common/model/Cacao.kt

```
@@ -4,11 +4,12 @@ package com.walletconnect.auth.common.model
 
 import com.squareup.moshi.Json
 import com.squareup.moshi.JsonClass
+import com.walletconnect.auth.signature.CacaoType
 
 @JsonClass(generateAdapter = true)
 internal data class Cacao(
     @Json(name = "header")
-    val header: Header,
+    val header: Header = CacaoType.EIP4361.toHeader(),
```

i think we should not default this property, since it hides the business logic in the model instead of the engine